server/src/main/java/io/spine/server/ServerEnvironment.java

```diff
@@ -48,25 +49,30 @@
 /**
  * The server conditions and configuration under which the application operates.
  *
- * <h1>Configuration</h1>
+ * <h3>Configuration</h3>
+ * 
  * <p>Some parts of the {@code ServerEnvironment} can be customized based on the {@code
  * EnvironmentType}. To do so, one of the overloads of the {@code use} method can be called.
- * Two environment types exist out of the box: {@link Tests} and {@link Production}. For example:
+ * Two environment types exist out of the box: {@link Tests} and {@link Production}.
+ * For example:
  * <pre>
- *     ServerEnvironment.use(productionStorageFactory, Production.class)
- *                      .use(testingStorageFactory, Tests.class)
- *                      .use(memoizingTracerFactory, Production.class)
+ *
+ *     ServerEnvironment.when(Production.class)
+ *                      .use(productionStorageFactory)
+ *                      .use(memoizingTracerFactory);
+ *     ServerEnvironment.when(Tests.class)
+ *                      .use(testingStorageFactory);
  * </pre>
- * A custom environment type may also be used:
+ *
+ * <p>A custom environment type may also be used:
  * <pre>
+ *
  *     final class StagingEnvironment extends EnvironmentType {
  *         ...
  *     }
  *
- *     // Can also be
- *     // `ServerEnvironment.use(inMemoryStorageFactory, new Staging(stagingServiceDiscovery));`,
```

Comment:
Why do we drop API which allows passing of `EnvironmentType`-s with dependencies, like the one here? Usage looks more or less realistic to me.

Reply:
It's going to be addressed in the following PR by introducing `use(Function<Class<? extends EnvironmentType>>, V>)`.